src/internal/observable/combineLatest.ts

```diff
@@ -9,7 +9,17 @@ import { Subscription } from '../Subscription';
 import { mapOneOrManyArgs } from '../util/mapOneOrManyArgs';
 import { popResultSelector, popScheduler } from '../util/args';
 
-/* tslint:disable:max-line-length */
+// combineLatest([a, b, c])
+export function combineLatest(sources: []): Observable<never>;
+export function combineLatest<A extends readonly unknown[]>(sources: readonly [...ObservableInputTuple<A>]): Observable<A>;
+
+// combineLatest(a, b, c)
+/** @deprecated Use the version that takes an array of Observables instead */
+export function combineLatest<A extends readonly unknown[]>(...sources: [...ObservableInputTuple<A>]): Observable<A>;
+
+// combineLatest({a, b, c})
+export function combineLatest(sourcesObject: {}): Observable<never>;
```

Comment:
issue: `{}` is not the type for an empty object. It will match anything that's neither `null` nor `undefined`. Instead, this should be:

```ts
export function combineLatest(sourcesObject: { [K in any]: never }): Observable<never>;
```

This should be fixed in the `forkJoin` signature, too. You might even want to declare an `EmptyObject` type for this, so that things are a little clearer.